src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.ITriviaDataGetter.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    internal partial class TokenStream
+    {
+        private interface ITriviaDataGetter
+        {
+            TriviaData GetTriviaData(TokenData token1, TokenData token2);
+        }
```

Comment:
note: this is great.   I would have also been ok with you keeping all three nested types together in a separate file if you liked that.   Just for future reference.